### DISCUSSION TITLE
TypeScript definitions

### DISCUSSION THREAD

**Discussion by mitchell-merry** [Noxcrew/mcchampionship-api#2] [Q&A] [answered]
Do any (public) typescript definitions exist for this API?

Edit: If they don't I'd start work on making them, to clarify.

**Comment by MagicA550** [comment, marked as answer]
Hey there!

Thanks for your question. The easiest way to get types is to generate it yourself using [openapi-typescript-codegen](https://www.npmjs.com/package/openapi-typescript-codegen) which automatically creates a HTTP client in TypeScript using the [OpenAPI schema](https://api.mcchampionship.com/openapi.json). Although I believe there are some wrappers already in progress on NPM from third parties you're always welcome to work on one yourself unofficially.

**Reply by mitchell-merry** [reply]
Thanks anthony, do you have any links to some of the wrappers?

**Reply by mitchell-merry** [reply]
Nevermind, found niklaas' API! https://www.npmjs.com/package/node-mcc-api
